### ISSUE TITLE
[bug] dyld: Library not loaded when using npm run tauri dev

### ISSUE THREAD

**Issue by cheng-lim** [tauri-apps/tauri#9084]
## Error
Got this error message when running `npm run tauri dev`.
```
dyld[69471]: Library not loaded: @rpath/libiconv.2.dylib
  Referenced from: <6C1414E8-0766-33FF-96BB-C5A26AC73B61> /Users/foo/Documents/workspace/foo/src-tauri/target/debug/foo
  Reason: no LC_RPATH's found
```
Similar to the error message in this [issue](https://github.com/tauri-apps/tauri/issues/7547). However, I am able to launch the backend app under `src-tauri` directory using `cargo run` without encountering this issue.

## Tauri overrides
I've tried to manually assign the correct path to `RPATH` by using the following command.
```
install_name_tool -add_rpath /usr/local/opt/libiconv/lib /Users/foo/Documents/workspace/foo/src-tauri/target/debug/foo
```
I can confirm it was successfully inserted.
```
otool -l /Users/foo/Documents/workspace/foo/src-tauri/target/debug/foo | grep -A2 LC_RPATH
          cmd LC_RPATH
      cmdsize 40
         path /usr/local/opt/libiconv/lib (offset 12)
```
         
However, when I ran `npm run tauri dev`, above's error message appeared. To confirm my modification again,
```
otool -l /Users/foo/Documents/workspace/foo/src-tauri/target/debug/foo | grep -A2 LC_RPATH
          cmd LC_RPATH
      cmdsize 40
         path /usr/local/opt/libiconv/lib (offset 12)
```
it returned nothing. My modification is gone. It seems like tauri has overridden the modification?

### I encounter this issue with the [OasysDB](https://github.com/oasysai/oasysdb) crate.
1. cargo add oasysdb   
2. npm run tauri dev

### Expected behavior

`npm run tauri dev` launches the app without issue.

### Full `tauri info` output

```text
[✔] Environment
    - OS: Mac OS 14.3.1 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.76.0 (07dca489a 2024-02-04) (Homebrew)
    ✔ cargo: 1.76.0
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-apple-darwin (default)
    - node: 20.5.0
    - yarn: 1.22.19
    - npm: 8.1.3

[-] Packages
    - tauri [RUST]: 1.5.4
    - tauri-build [RUST]: 1.5.1
    - wry [RUST]: 0.24.7
    - tao [RUST]: 0.16.6
    - tauri-cli [RUST]: 1.5.9
    - @tauri-apps/api [NPM]: 1.5.3
    - @tauri-apps/cli [NPM]: 1.5.10

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: React
    - bundler: Vite
```

### Stack trace

```text
dyld[69471]: Library not loaded: @rpath/libiconv.2.dylib
  Referenced from: <6C1414E8-0766-33FF-96BB-C5A26AC73B61> /Users/foo/Documents/workspace/foo/src-tauri/target/debug/foo
  Reason: no LC_RPATH's found
```

**Comment by FabianLars** [comment]
Thanks for the detailed report!

Can you check if `cargo run` (inside src-tauri) works fine? If it does, then i think this is the same as https://github.com/tauri-apps/tauri/issues/4724

**Comment by cheng-lim** [comment]
Yes, it works fine inside src-tauri.

**Comment by cheng-lim** [comment]
The solution in #4724 didn't work for me.
However, I managed to solve this issue by modifying the `src-tauri/build.rs` to the following code.
```rust
fn main() {
  tauri_build::build();
  #[cfg(target_os = "macos")]
  {
    println!("cargo:rustc-link-arg=-Wl,-rpath,/usr/local/opt/libiconv/lib");
  }
}

```

**Comment by FabianLars** [comment]
> The solution in https://github.com/tauri-apps/tauri/issues/4724 didn't work for me.

Yeah, they were talking about Linux only. It just has the same root cause, but not the same solution :/

Cool you found a workaround for now though! Also, closing this as a duplicate of 4724 then :)

**Comment by ivnsch** [comment]
https://github.com/tauri-apps/tauri/issues/9084#issuecomment-1978876659 also fixed it for me ([#4724](https://github.com/tauri-apps/tauri/issues/4724) didn't work either)

**Comment by PJUllrich** [comment]
I had a similar issue today and wanted to share how I debugged and fixed it.

When I started my macOS library on another Mac that didn't have `homebrew` installed, it crashed upon startup with this issue:

```
Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: /opt/homebrew/*/liblzma.5.dylib
Referenced from: <...> /Applications/My App.app/Contents/MacOS/my-app
Reason: tried: '/opt/homebrew/*/liblzma.5.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/*/liblzma.5.dylib' (no such file), '/opt/homebrew/*/liblzma.5.dylib' (no such file)
```

So, the problem was that the `liblzma.5.dylib` didn't exist on the other system. 

First, I wanted to find out which rust library dependent on that library, so I ran this command:

```
cargo tree | grep -i lzma
```

I found out that my `zip` dependency used the `xz` library which used `lzma-rs` and that needed `liblzma.5.dylib`. It turned out that I I forgot to disable the optional features in my `zip` dependency.

At this point, I simply added only the feature to my `zip` dependency that I needed (`deflate`) which removed the `xz` feature and the `xz` and `lzma-rs` dependencies with it. 

```
zip = { version = "4.3.0", default-features = false, features = ["deflate"] }
```

I built the release locally and checked with the `otool -l` command that the `liblzma.4.dylib` dependency was gone and that fixed my issue!
